analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -115,7 +115,9 @@ private static void CheckGetTypeAndTypeOfEquality(SonarSyntaxNodeReportingContex
 
         private static void CheckForIsInstanceOfType(SonarSyntaxNodeReportingContext context, MemberAccessExpressionSyntax memberAccess, IMethodSymbol methodSymbol)
         {
-            if (methodSymbol.Name == "IsInstanceOfType" && memberAccess.Expression is TypeOfExpressionSyntax)
+            if (methodSymbol.Name == nameof(Type.IsInstanceOfType)
+                && memberAccess.Expression is TypeOfExpressionSyntax typeOf
+                && !(context.SemanticModel.GetTypeInfo(typeOf.Type).Type is INamedTypeSymbol { IsUnboundGenericType: true }))
```

Comment:
This used to be a syntax-only check. And we now query a semantic model that is expensive. I have a feeling that detecting unbound generic should be a simple thing to do => please try to do this on syntax level.

Reply:
Done! Indeed, it was as easy as looking for a descendant of type `OmittedTypeArgument`.

Same complexity, better performance!